rasa/core/brokers/pika.py

```diff
@@ -249,67 +251,45 @@ def close_pika_connection(connection: "Connection") -> None:
+Message = Tuple[Text, MessageHeaders]
+
+
+class PikaMessageProcessor:
+    """A class that holds all the Pika connection details and processes Pika messages.
+    All the messages should be published to a `process_queue` provided."""
 
     def __init__(
         self,
-        host: Text,
-        username: Text,
-        password: Text,
-        port: Union[int, Text] = 5672,
-        queues: Union[List[Text], Tuple[Text], Text, None] = None,
-        should_keep_unpublished_messages: bool = True,
-        raise_on_failure: bool = False,
-        log_level: Union[Text, int] = os.environ.get(
-            ENV_LOG_LEVEL_LIBRARIES, DEFAULT_LOG_LEVEL_LIBRARIES
-        ),
-        **kwargs: Any,
-    ):
-        """Initialise RabbitMQ event broker.
+        parameters: "Parameters",
+        get_message: Callable[[], Message],
+        queues: Union[List[Text], Tuple[Text], Text, None],
+    ) -> None:
+        """Initialise Pika connector.
 
         Args:
-            host: Pika host.
-            username: Username for authentication with Pika host.
-            password: Password for authentication with Pika host.
-            port: port of the Pika host.
-            queues: Pika queues to declare and publish to.
-            should_keep_unpublished_messages: Whether or not the event broker should
-                maintain a queue of unpublished messages to be published later in
-                case of errors.
-            raise_on_failure: Whether to raise an exception if publishing fails. If
-                `False`, keep retrying.
-            log_level: Logging level.
+            parameters: Pika connection parameters
+            queues: Pika queues to declare and publish to
         """
-        logging.getLogger("pika").setLevel(log_level)
+        self.parameters: "Parameters" = parameters
+        self.queues: Union[List[Text], Tuple[Text]] = self._get_queues_from_args(queues)
+        self.get_message: Callable[[], Message] = get_message
 
-        self.host = host
-        self.username = username
-        self.password = password
-        self.port = port
+        self.pika_connection: Optional["SelectConnection"] = None
         self.channel: Optional["Channel"] = None
-        self.queues = self._get_queues_from_args(queues)
-        self.should_keep_unpublished_messages = should_keep_unpublished_messages
-        self.raise_on_failure = raise_on_failure
-
-        # List to store unpublished messages which hopefully will be published later
-        self._unpublished_messages: Deque[Text] = deque()
-        self._run_pika()
 
     def __del__(self) -> None:
         if self.channel:
             close_pika_channel(self.channel)
             close_pika_connection(self.channel.connection)
 
+        if self.is_connected:
+            self.pika_connection.close()
+
     def close(self) -> None:
-        """Close the pika channel and connection."""
+        """Close the Pika connector."""
```

Comment:
```suggestion
        """Close the Pika connection."""
```
since it's calling `close_pika_connection()` no?